app/controllers/line_items_controller.rb

```diff
@@ -1,11 +1,33 @@
 class LineItemsController < ApplicationController
+  include CurrentCart
+  before_action :set_cart, only: [:create]
 
   def new
     @line_item = LineItem.new
   end
 
   def create
-    @line_item = LineItem.create(line_item_params)
+      product = Product.find(params[:product_id])
+      @product_line_item = cart.line_items.find_by(product: product)
+      if @product_line_item && params[:line_item].present?
+        @product_line_item.update_attribute(quantity: @product_line_item.quantity + line_item_params[:quantity])
+        redirect_to root_path
```

Comment:
as all scenarios redirects to root path, you can remove it from all of them and just put it once after the if clause